clang/include/clang/Driver/Options.td

```diff
@@ -8545,6 +8545,11 @@ def dxc_entrypoint : Option<["--", "/", "-"], "E", KIND_JOINED_OR_SEPARATE>,
                      Group<dxc_Group>,
                      Visibility<[DXCOption]>,
                      HelpText<"Entry point name">;
+def dxc_hlsl_version : Option<["/", "-"], "HV", KIND_JOINED_OR_SEPARATE>,
+                     Group<dxc_Group>,
+                     Visibility<[DXCOption]>,
+                     HelpText<"HLSL Version">,
+                     Values<"2016, 2017, 2018, 2021, 202x">;
```

Comment:
Does the Values already make sure error be reported when give input not in the Values?

Reply:
No, I added that logic myself.  I've manually checked all the valid values and emit an error if there are no matches in  HLSL.cpp:263.
Regardless, I wouldn't need that logic, because I need to perform a unique transformation exhaustively in every valid case, and if there are no matching valid cases, then I emit an error with `else`.

Reply:
I think the problem is that you're using `Values` wrong. That should be taking a list of possible values not a string.

Look at how the `target_profile` flag is implemented for an example.

Reply:
Oh! Also look at some of the other options. You can use a combination of `Values`, `NormalizedValues` and `MarshallingInfoEnum`, to have the parser generated to parse directly to the `HLSLLangStd` enumeration.

Reply:
I've decided to use NormalizedValues instead of values, but I no longer need access to valid values thanks to Xiang's suggestion. So I think I should just leave the values there for documentation's sake, but the values won't actually be read by option translation.